### PR TITLE
Add non-context manager usage support for profiler

### LINKED ISSUE #60238: [torch.profiler] non-context manager use
## 🚀 Feature

I need to embed the profiler around a huge training loop and to make it optional. So context manager isn't a good option here. And there are already some 5 indentation levels for the code to be wrapped, so for the sake of readability I'd rather not use some noop context manager.

Would it be possible to expose the enter/exit API as public (probably as start/stop) so that one could do:
```
if profile_enable:
    p = profiler.start()
normal multi-line code here
if profile_enable:
    p.stop()
```

Thank you!

@gaoteng-git 

cc @ilia-cher @gdankel @ngimel @nbcsm @guotuofeng @guyang3532 @gaoteng-git

### PR DESCRIPTION
**Pull request by guotuofeng**
Fixes https://github.com/pytorch/pytorch/issues/60238, https://github.com/pytorch/kineto/issues/329

### CODE REVIEW

**Comment by gdankel** [comment]
@ilia-cher and @ngimel would like to hear your opinion on this approach.

**Comment by ngimel** [comment]
This approach lgtm. I don't know what's up with windows kineto build failures though, is it bad base revision?

File: test/test_profiler.py [hunk 1]
```diff
@@ -559,7 +559,7 @@ def test_tensorboard_trace_handler(self):
 
         # test case for gzip file format
         with TemporaryDirectoryName() as dname:
-            with profile(
```
Comment (gdankel) [inline test/test_profiler.py:562]: Don't we also want to keep the old context manager based test?

Comment (ngimel) [reply test/test_profiler.py:562]: Definitely! We should have both context-manager and non-context-manager tests

Comment (guotuofeng) [reply test/test_profiler.py:562]: there are two test in this case. Line 533 use the context manager way and line 562 use the non-context manager ways

**Comment by guotuofeng** [comment]
> This approach lgtm. I don't know what's up with windows kineto build failures though, is it bad base revision?

I merged the latest changes from master. Let's see the build status

**Review by ngimel — APPROVED** [review]

**Comment by stas00** [comment]
Thank you for implementing this, @guotuofeng

So can we start using it in pt-1.10.0?

Could you please also update the documentation? Thank you!

**Comment by guotuofeng** [comment]
> Thank you for implementing this, @guotuofeng
> 
> So can we start using it in pt-1.10.0?
> 
> Could you please also update the documentation? Thank you!

will update the document and update the status here.

**Comment by guotuofeng** [comment]
> Thank you for implementing this, @guotuofeng
> 
> So can we start using it in pt-1.10.0?
> 
> Could you please also update the documentation? Thank you!

The documentation PR is https://github.com/pytorch/tutorials/pull/1625

**Comment by stas00** [comment]
Excellent! Thank you very much for adding it, @guotuofeng

**Pull request closed without merging**